service/state/core.go

```diff
@@ -127,20 +84,13 @@ func NewCore(dir string) (*Core, error) {
 		err = db.Update(func(tx *badger.Txn) error {
 
 			// store the empty root hash as last commit
-			key := []byte{prefixLastCommit}
-			err = tx.Set(key, commit)
+			err = SetLastCommit(commit)(tx)
```

Comment:
I think we should agree on a naming scheme here. `Set` seems not great; I would prefer `SaveLastCommit`.